### ISSUE TITLE
test: TestFuncTxnProduceNoBegin flakey, doesn’t handle/ignore/retry-after loading of offsets

### ISSUE THREAD

**Issue by dnwe** [IBM/sarama#2529]
TestFuncTxnProduceNoBegin periodically fails if the leader is still loading offsets:
```go
TestFuncTxnProduceNoBegin is #2 (localhost:29092)
2023-07-28T09:13:52.8501522Z [sarama] 2023/07/28 09:13:52 txnmgr/transition [TestFuncTxnProduceNoBegin] transition from ProducerTxnStateUninitialized to ProducerTxnStateInError|ProducerTxnStateFatalError
2023-07-28T09:13:52.8510148Z     functional_producer_test.go:113: 
2023-07-28T09:13:52.8510873Z         	Error Trace:	github.com/IBM/sarama/functional_producer_test.go:113
2023-07-28T09:13:52.8511456Z         	Error:      	Received unexpected error:
2023-07-28T09:13:52.8513039Z         	            	kafka server: The broker is still loading offsets after a leader change for that offset's topic partition
2023-07-28T09:13:52.8513678Z         	Test:       	TestFuncTxnProduceNoBegin
2023-07-28T09:13:52.8524719Z --- FAIL: TestFuncTxnProduceNoBegin (1.23s)
```

**Comment by dnwe** [comment]
Not seen this happen since the associated PR was merged, closing

**Comment by dnwe** [comment]
Failed just now in https://github.com/IBM/sarama/actions/runs/6012746470/job/16308708213?pr=2627 (see [latest.txt](https://github.com/IBM/sarama/files/12464998/latest.txt))

```
--- FAIL: TestFuncTxnProduceNoBegin (1.20s)

[DEBUG] 13:18:54.644896 Initializing new client
[DEBUG] 13:18:54.645123 Successfully initialized new client
[DEBUG] 13:18:54.645912 Connected to broker at 127.0.0.1:29094 (unregistered)
[DEBUG] 13:18:54.652176 Closed connection to broker 127.0.0.1:29094
[DEBUG] 13:18:54.652324 Closing Client
[DEBUG] 13:18:54.652748 Initializing new client
[DEBUG] 13:18:54.652826 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[DEBUG] 13:18:54.652970 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[DEBUG] 13:18:54.653667 Connected to broker at 127.0.0.1:29094 (unregistered)
[DEBUG] 13:18:54.654042 client/metadata fetching metadata for all topics from broker 127.0.0.1:29094
[DEBUG] 13:18:54.661022 client/brokers registered new broker #5 at localhost:29095
[DEBUG] 13:18:54.661100 client/brokers registered new broker #4 at localhost:29094
[DEBUG] 13:18:54.661135 client/brokers registered new broker #1 at localhost:29091
[DEBUG] 13:18:54.661191 client/brokers registered new broker #2 at localhost:29092
[DEBUG] 13:18:54.661222 client/brokers registered new broker #3 at localhost:29093
[DEBUG] 13:18:54.662561 Successfully initialized new client
[DEBUG] 13:18:54.662658 txnmgr/init-producer-id [TestFuncTxnProduceNoBegin] invoking InitProducerId with current producer ID -1 and epoch -1 in order to bump the epoch
[DEBUG] 13:18:54.662725 client/coordinator requesting coordinator for TestFuncTxnProduceNoBegin from 127.0.0.1:29094
[DEBUG] 13:18:54.772046 client/coordinator coordinator for TestFuncTxnProduceNoBegin is not available
[DEBUG] 13:18:54.772281 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[DEBUG] 13:18:54.772478 client/metadata fetching metadata for [__transaction_state] from broker 127.0.0.1:29094
[DEBUG] 13:18:54.773782 Connected to broker at localhost:29094 (registered as #4)
[DEBUG] 13:18:54.796503 client/metadata found some partitions to be leaderless
[DEBUG] 13:18:55.047279 client/metadata retrying after 250ms... (2 attempts remaining)
[DEBUG] 13:18:55.047487 client/metadata fetching metadata for [__transaction_state] from broker 127.0.0.1:29094
[DEBUG] 13:18:55.049811 client/metadata found some partitions to be leaderless
[DEBUG] 13:18:55.300936 client/metadata retrying after 250ms... (1 attempts remaining)
[DEBUG] 13:18:55.301160 client/metadata fetching metadata for [__transaction_state] from broker 127.0.0.1:29094
[DEBUG] 13:18:55.309637 client/metadata found some partitions to be leaderless
[DEBUG] 13:18:55.560038 client/metadata retrying after 250ms... (0 attempts remaining)
[DEBUG] 13:18:55.560104 client/metadata fetching metadata for [__transaction_state] from broker 127.0.0.1:29094
[DEBUG] 13:18:55.574788 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
[DEBUG] 13:18:55.575057 client/coordinator retrying after 250ms... (2 attempts remaining)
[DEBUG] 13:18:55.576040 Connected to broker at localhost:29091 (registered as #1)
[DEBUG] 13:18:55.825379 client/coordinator requesting coordinator for TestFuncTxnProduceNoBegin from 127.0.0.1:29094
[DEBUG] 13:18:55.828478 client/coordinator coordinator for TestFuncTxnProduceNoBegin is #1 (localhost:29091)
[DEBUG] 13:18:55.839448 txnmgr/transition [TestFuncTxnProduceNoBegin] transition from ProducerTxnStateUninitialized to ProducerTxnStateInError|ProducerTxnStateFatalError
    functional_producer_test.go:112: 
        	Error Trace:	github.com/IBM/sarama/functional_producer_test.go:112
        	Error:      	Received unexpected error:
        	            	kafka server: The broker is still loading offsets after a leader change for that offset's topic partition
        	Test:       	TestFuncTxnProduceNoBegin
  ```
